pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1803,6 +1815,36 @@ public void addTable(TableConfig tableConfig)
     LOGGER.info("Adding table {}: Successfully added table", tableNameWithType);
   }
 
+  /**
+   * Adds a logical table.
+   * @param logicalTableConfig The logical table config to be added
+   * @throws TableAlreadyExistsException If the logical table already exists
+   */
+  public void addLogicalTableConfig(LogicalTableConfig logicalTableConfig)
+      throws TableAlreadyExistsException {
+    String tableName = logicalTableConfig.getTableName();
+    LOGGER.info("Adding logical table {}: Start", tableName);
+
+    // Check if the logical table name is already used
+    if (ZKMetadataProvider.isLogicalTableExists(_propertyStore, tableName)) {
+      throw new TableAlreadyExistsException("Logical table: " + tableName + " already exists");
+    }
+
+    // Check if the table name is already used by a physical table
+    getAllTables().stream().map(TableNameBuilder::extractRawTableName).distinct().filter(tableName::equals)
```

Comment:
A much cheaper way is to check if the 2 corresponding table names (with `_OFFLINE` and `_REALTIME`) exists.

We want to add the same check when adding physical table to avoid collision with logical table name.

Reply:
Addressed here: https://github.com/apache/pinot/pull/15720/commits/22cc927f77dc967f8207fbb76d0e17012145d098